### PR TITLE
Add cpu-model feature detection

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>
First pass on enhancing the discovered model of the CPU
example:
```bash
                    feature.node.kubernetes.io/cpu-model.ID=113
                    feature.node.kubernetes.io/cpu-model.family=23
                    feature.node.kubernetes.io/cpu-model.vendorID=AMD
```

### CODE REVIEW

File: source/cpu/cpu.go [hunk 1]
```diff
@@ -221,6 +232,15 @@ func (s *cpuSource) GetFeatures() *feature.DomainFeatures {
 	return s.features
 }
 
+func getCPUModel() map[string]string {
+	cpuModelInfo := make(map[string]string)
+	cpuModelInfo["vendorID"] = cpuid.CPU.VendorID.String()
+	cpuModelInfo["family"] = strconv.Itoa(cpuid.CPU.Family)
+	cpuModelInfo["ID"] = strconv.Itoa(cpuid.CPU.Model)
```
Comment (marquiz) [inline source/cpu/cpu.go:239]: nit: I wouldn't probably use `camelCase` in the labels (and not `CamelCase` either). If something, I think, the prevailing style so far is `snake_case` so I'd suggest to use it. WDYT?
 
```suggestion
	cpuModelInfo["vendor_id"] = cpuid.CPU.VendorID.String()
	cpuModelInfo["family"] = strconv.Itoa(cpuid.CPU.Family)
	cpuModelInfo["id"] = strconv.Itoa(cpuid.CPU.Model)
```

**Review by marquiz — COMMENTED** [review]
Thanks @ArangoGutierrez!

Could you also update the docs (featues.md and customization-guide.md)?

File: docs/get-started/features.md [hunk 2]
```diff
@@ -51,6 +51,9 @@ such as restricting discovered features with the -label-whitelist option.*
 | **`cpu-cstate.enabled`**          | bool   | Set to 'true' if cstates are set in the intel_idle driver, otherwise set to 'false'. Unset if intel_idle cpuidle driver is not active.
 | **`cpu-rdt.<rdt-flag>`**          | true   | [Intel RDT][intel-rdt] capability is supported. See [RDT flags](#intel-rdt-flags) for details.
 | **`cpu-sgx.enabled`**             | true   | Set to 'true' if Intel SGX is enabled in BIOS (based a non-zero sum value of SGX EPC section sizes).
+| **`cpu-model.vendor_id`**             | string   | Comparable CPU vendor ID.
+| **`cpu-model.family`**             | int   | CPU family model.
+| **`cpu-model.id`**             | int   | CPU model number.
```
Comment (marquiz) [inline docs/get-started/features.md:56]: nit: fixing the alignment of the cells would make this more readable also in textfile form

File: docs/get-started/features.md [hunk 3]
```diff
@@ -51,6 +51,9 @@ such as restricting discovered features with the -label-whitelist option.*
 | **`cpu-cstate.enabled`**          | bool   | Set to 'true' if cstates are set in the intel_idle driver, otherwise set to 'false'. Unset if intel_idle cpuidle driver is not active.
 | **`cpu-rdt.<rdt-flag>`**          | true   | [Intel RDT][intel-rdt] capability is supported. See [RDT flags](#intel-rdt-flags) for details.
 | **`cpu-sgx.enabled`**             | true   | Set to 'true' if Intel SGX is enabled in BIOS (based a non-zero sum value of SGX EPC section sizes).
+| **`cpu-model.vendor_id`**             | string   | Comparable CPU vendor ID.
+| **`cpu-model.family`**             | int   | CPU family model.
```
Comment (marquiz) [inline docs/get-started/features.md:55]: "CPU family model"?? Just "CPU family" or "CPU model family" perhaps?

File: v10.yaml [hunk 4]
```diff
@@ -0,0 +1,389 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: node-feature-discovery
+---
```
Comment (marquiz) [inline v10.yaml:5]: What on earth is this file? 🚀  😆

File: docs/advanced/customization-guide.md [hunk 5]
```diff
@@ -493,6 +493,7 @@ The following features are available for matching:
 | **`cpu.sst`**    | attribute    |          |            | Intel SST (Speed Select Technology) capabilities
 |                  |              | **`bf.enabled`** | bool | `true` if Intel SST-BF (Intel Speed Select Technology - Base frequency) has been enabled, otherwise does not exist
 | **`cpu.topology`** | attribute  |          |            | CPU topology related features
+| **`cpu.model`** | attribute  |          |            | CPU model related attributes
```
Comment (marquiz) [inline docs/advanced/customization-guide.md:496]: nit: the "Feature" column should be in alphabetical order (I hope it is atm) so move this before `cpu.pstate`.

Also, could you list the available "elements" (i.e. family, vendor_id and id), similar how they are listed with other features (like `cpu.pstate`)

File: v10.yaml (hunk 4 above)
Comment (ArangoGutierrez) [reply v10.yaml:5]: hahahaha ups , `git add .` got me deleting.....

File: source/cpu/cpu.go [hunk 6]
```diff
@@ -141,6 +144,11 @@ func (s *cpuSource) GetLabels() (source.FeatureLabels, error) {
 		}
 	}
 
+	// CPUMODEL
```
Comment (marquiz) [inline source/cpu/cpu.go:147]: nit:
```suggestion
	// CPU model
```

**Review by marquiz — APPROVED** [review]
Looks good to me now 😎 

/label tide/merge-method-squash

**Comment by marquiz** [comment]
@zvonkok PTAL

**Comment by zvonkok** [comment]
@marquiz can we use Arm based instances with prow? Since Arm is going to be a major player in the near term, I would like to make sure we're testing x86 and Arm.

**Comment by zvonkok** [comment]
Let me check on a use-case if we might extend the list of feature we have here.

**Comment by ArangoGutierrez** [comment]
> @marquiz can we use Arm based instances with prow? Since Arm is going to be a major player in the near term, I would like to make sure we're testing x86 and Arm.

>https://kubernetes.slack.com/archives/C09QZ4DQB/p1648205765501509

So Only Build-X for now

**Comment by marquiz** [comment]
> So only Build-X for now

Yeah, that was my understanding, too. Thanks for checking this @ArangoGutierrez.


> Let me check on a use-case if we might extend the list of feature we have here.

I think we could just merge this one and extend support in future PRs if something is missing(?)

**Pull request merged**